src/Microsoft.Extensions.Configuration.AzureAppConfiguration/JsonConfigurationParser.cs

```diff
@@ -0,0 +1,108 @@
+                            if (Array.Exists(subTypes, x => x == acceptedSubType))
+                            {
+                                return true;
+                            }
+                        }
+                        else if (Equals(subType, acceptedSubType))
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+            catch (Exception ex) when (ex is ArgumentException || ex is FormatException || ex is NullReferenceException)
+            {
+                // not a valid JSON content type
+            }
+            return false;
+        }
+
+        public static void ParseJsonSetting(ConfigurationSetting setting, IDictionary<string, ConfigurationSetting> data)
+        {
+            SortedDictionary<string, string> keyValueDict = new SortedDictionary<string, string>();
+            ParseSetting(setting.Key, setting.Value, keyValueDict);
+            foreach (KeyValuePair<string, string> entry in keyValueDict)
+            {
+                ConfigurationSetting newSetting = new ConfigurationSetting(entry.Key, entry.Value, setting.Label);
+                newSetting.ContentType = setting.ContentType;
+                data[newSetting.Key] = newSetting;
+            }
+        }
+
+
+        public static void ParseSetting(string currentKey, string currentValue, SortedDictionary<string, string> keyValueDict)
+        {
+            try
+            {
+                var json_data = JsonSerializer.Deserialize<JsonElement>(currentValue);
+                switch (json_data.ValueKind)
+                {
+                    case JsonValueKind.Array:
+                        var valueArray = json_data.EnumerateArray();
+                        for (int index = 0; index < valueArray.Count(); index++)
+                        {
+                            var newKey = ConfigurationPath.Combine(new List<string> { currentKey, index.ToString() });
+                            ParseSetting(newKey, valueArray.ElementAt(index).GetRawText(), keyValueDict);
+                        }
+                        break;
+
+                    case JsonValueKind.Object:
+                        var valueObject = json_data.EnumerateObject();
+                        foreach(JsonProperty entry in valueObject)
+                        {
+                            var newKey = ConfigurationPath.Combine(new List<string> { currentKey, entry.Name.ToString() });
+                            ParseSetting(newKey, entry.Value.GetRawText(), keyValueDict);
+                        }
+                        break;
+
+                    default:
+                        keyValueDict[currentKey] = json_data.ToString();
+                        break;
+                }
+
+            }
+            catch (Exception ex) when (ex is ArgumentNullException || ex is JsonException || ex is NotSupportedException)
```

Comment:
>ArgumentNullException [](start = 45, length = 21)

Where do expect this exception to be thrown?

Reply:
From the docs, JsonSerializer.Deserialize can throw these three exceptions: [https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserialize?view=netcore-3.1](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserialize?view=netcore-3.1) 

I can reproduce ArgumentNullException when the a setting has null value and json content type. For example, this setting was created through CLI:
Az appconfig kv set --key foobar --content-type application/json -y
{
  "contentType": "application/json",
  "etag": "RAHNsqogncZZdEIkBgyTM1in7FT",
  "key": "foobar",
  "label": null,
  "lastModified": "2020-07-13T19:50:20+00:00",
  "locked": false,
  "tags": {},
  "value": null
}

Notice that the value is null. Since the value is invalid JSON, going forward, we will not allow users to create new settings with null value and JSON content type, but there could be some customers with existing settings like this.

I was not able to reproduce NotSupportedException but still kept it to be on the safer side.


---
In reply to: [453141145](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r453141145) [](ancestors = 453141145)

Reply:
We can update `CanProcess` to return `false` if the value is `null`. Then we wouldn't need to handle it in the parser.

---
In reply to: [453895459](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r453895459) [](ancestors = 453895459,453141145)

Reply:
Done

---
In reply to: [454037685](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r454037685) [](ancestors = 454037685,453895459,453141145)